packaging/irods_prc_tests/test_rule_engine_plugin_metadata_guard_atomic.py

```diff
@@ -0,0 +1,145 @@
+            self.assertEqual(obj.metadata['unguarded::atr1'], iRODSMeta('unguarded::atr1', 'val1'))
+
+            # "irods::" protected, so metadata add should fail
+            self.assertRaises(irods.exception.CAT_INSUFFICIENT_PRIVILEGE_LEVEL, lambda: obj.metadata.apply_atomic_operations( AVUOperation(operation='add', avu=iRODSMeta('irods::atr','val'))))
+
+            # "irods::" protected, so metadata delete should fail
+            self.assertRaises(irods.exception.CAT_INSUFFICIENT_PRIVILEGE_LEVEL, lambda: obj.metadata.apply_atomic_operations( AVUOperation(operation='remove', avu=iRODSMeta('irods::adminonly','adminonly'))))
+
+    def test_guard_atomic_operations_editor_list__issue_38(self):
+        # set metadata_guard config AVU on collection to admin_only: false and irods:: as protected prefix
+        # also, add test user as editor
+        with iRODSSession(host='localhost', port=1247, user='rods', password='rods', zone='tempZone') as local_admin:
+            coll = local_admin.collections.get("/tempZone")
+            METADATA_GUARD_CONFIG = {
+                'admin_only': False,
+                'editors': [{'name': 'rods', 'type': 'user'}, {'name': 'alice', 'type': 'user'}],
+                'prefixes': ['irods::']
+            }
+            coll.metadata['irods::metadata_guard'] = iRODSMeta('irods::metadata_guard', json.dumps(METADATA_GUARD_CONFIG))
+
+        with iRODSSession(host='localhost', port=1247, user='alice', password='test', zone='tempZone') as unprivileged_user:
+            obj = unprivileged_user.data_objects.get(self.TEST_DATA_OBJECT_PATH)
+
+            # operation should succeed, as alice is set as an editor
+            obj.metadata.apply_atomic_operations( AVUOperation(operation='add', avu=iRODSMeta('irods::editorlist', 'editorlist')))
+            self.assertEqual(obj.metadata['irods::editorlist'], iRODSMeta('irods::editorlist', 'editorlist'))
+
+        # remove test user from editor list
+        with iRODSSession(host='localhost', port=1247, user='rods', password='rods', zone='tempZone') as local_admin:
+            coll = local_admin.collections.get("/tempZone")
+            METADATA_GUARD_CONFIG = {
+                'admin_only': False,
+                'editors': [{'name': 'rods', 'type': 'user'}],
+                'prefixes': ['irods::']
+            }
+            coll.metadata['irods::metadata_guard'] = iRODSMeta('irods::metadata_guard', json.dumps(METADATA_GUARD_CONFIG))
+
+        with iRODSSession(host='localhost', port=1247, user='alice', password='test', zone='tempZone') as unprivileged_user:
+            obj = unprivileged_user.data_objects.get(self.TEST_DATA_OBJECT_PATH)
+
+            # operation should succeed, as alice is set as an editor
+            self.assertRaises(irods.exception.CAT_INSUFFICIENT_PRIVILEGE_LEVEL, lambda: obj.metadata.apply_atomic_operations( AVUOperation(operation='remove', avu=iRODSMeta('irods::editorlist', 'editorlist'))))
```

Comment:
`irods::editorlist` -> `irods::metadata_guard`?

Reply:
I think this follows line 124/125, where the avu irods::editorlist is added (so it can be deleted here), but I could have it add another, different one for clarity.

Reply:
It's fine. I hadn't considered the AVU being added much earlier in the test and the diff I was reviewing did not show the addition of the AVU.

Adding a comment here to remind the reader that the AVU was added earlier would be helpful. Alternatively, you could assert the existence of the AVU on the data object. Up to you.